src/app/services/graph-client/auth/authentication-module.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  AccountInfo,
+  AuthenticationResult, InteractionRequiredAuthError,
+  PopupRequest, PublicClientApplication, SilentRequest
+} from '@azure/msal-browser';
+
+import { geLocale } from '../../../../appLocale';
+import { AUTH_URL, DEFAULT_USER_SCOPES } from '../../graph-constants';
+
+const defaultScopes = DEFAULT_USER_SCOPES.split(' ');
+
+export class AuthenticationModule {
```

Comment:
how did you test that this is now doing the right thing as far as PKCE.

Reply:
@ddyett 
Frankly, I have trusted that the msal team has done the right thing. My tests were geared towards whether GE was able to sign in and out effectively into all the accounts. I may need guidance on ways to test this the PKCE way.

Reply:
@thewahome Turn on Fiddler, you'll see a call to /authorize with &code_challenge parameter. That indicates that it is using PKCE.  

https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow#request-an-authorization-code

You'll also see the refresh token supplied in the response to /token.

Graph Explorer and MSAL are using the auth code flow as expected. The one thing that we likely aren't doing but should do is set the state token in the initial request to /authorize and then checking on the response from /token that we are getting the same state token. This helps prevent CSRF